### ISSUE TITLE
Request decorator

### ISSUE THREAD

**Issue by JimOverholt** [cisco-open/cisco-catalyst-wan-sdk#306]
# Pull Request summary:
## TLDR;
This PR introduces `@request` decorator which is used to define endpoint handlers.
### Before:
```python
def find_status(self, task_id: str) -> TaskData:
    url = f"/device/action/status/{task_id}"
    return self._get(url).dataobj(TaskData, None)
```
### After:
```python
@request(get, "/device/action/status/{task_id}")
def find_status(self, task_id: str) -> TaskData:
    ...
```
## Benefit?
- less code, less errors
- documentation automatically generated with `pre-commit`: [ENDPOINTS.md](https://github.com/CiscoDevNet/vManage-client/blob/request-decorator/ENDPOINTS.md)
- no unittests needed for endpoints defined with decorator (method body predefined and unittested against all supported parameters, decorator and annotated method are checked during `pre-commit`)

# Description of changes:
Decorator inspects decorated method signature and arguments to specify payload and return type and to build request according to declaration. This decorator automatically builds a request during runtime so no method body is needed.
Various types of return values and payload are supported (Including CustomPayloadType)
When new endpoint is added in code, documentation is automatically generated by `pre-commit` in `ENDPOINTS.md` file. All existing APIEndpoints are migrated to use `@request` decorator in this PR.

`@request` decorator is implemented in: `vmngclient/endpoints/__init__.py`
To see all possible use (and misuse) cases check: `vmngclient/tests/test_endpoints.py`

## Full example of usage:
```python
# 1. Define endpoint using request decorator:
from pydantic import BaseModel, Field
from typing import List
from vmngclient.endpoints import APIEndpoints, request, versions, view, delete
from vmngclient.utils.session_type import ProviderView


class TenantBulkDeleteRequest(BaseModel):
    password: str
    tenant_id_list: List[str] = Field(alias="tenantIdList")


class TenantTaskId(BaseModel):
    id: str


class TenantManagement(APIEndpoints):
    @versions(">=20.4")
    @view({ProviderView})
    @request(delete, "/tenant/bulk/async")
    def delete_tenant_async_bulk(
        self, payload: TenantBulkDeleteRequest
    ) -> TenantTaskId:
        ...


# 2. Login to vManage with provider credentials
from vmngclient.session import create_vManageSession

session = create_vManageSession(
    url="example.com", username="provider", password="password123"
)

# 3. Instantiate endpoints class using session and send request
api = TenantManagement(session)
api.delete_tenant_async_bulk(
    TenantBulkDeleteRequest(password=session.password, tenantIdList=[])
)


```

# Checklist:
- [x] Make sure to run pre-commit before commiting changes
- [x] Make sure all checks have passed
- [x] PR description is clear and comprehensive
- [x] Mentioned the issue that this PR solves (if applicable)
- [x] Make sure you test the changes

**Comment by JimOverholt** [comment]
This issue was migrated from an old repository: https://github.com/CiscoDevNet/catalystwan/pull/305
